### PR TITLE
Add ~= selector

### LINKED ISSUE #35: global: support `~=` version specifier
Specifying requirements with `~=`:

```python
# setup.py
...
install_requires [
    'invenio-records~=1.0.0',
]
...
```
...gives the following error:

`ERROR: specify minimal version of "invenio-app" using ">=" or "=="`

### PR DESCRIPTION
**Pull request by greut**
closes #35, ping @slint

### CODE REVIEW

File: tests/test_cli.py [hunk 1]
```diff
@@ -21,119 +21,142 @@
 DATA = abspath(join(dirname(__file__), 'data/'))
 
 
-@pytest.fixture
+@pytest.fixture(scope="function")
```
Comment (jirikuncar) [inline tests/test_cli.py]: style comment: single quotes

**Comment by slint** [comment]
Package extras were not taken into account in the past (when building for `min`). I'm not sure if this should actually be considered a breaking change...

File: requirements_builder/requirements_builder.py [hunk 2]
```diff
@@ -169,8 +169,8 @@ def iter_requirements(level, extras, pip_file, setup_fp):
                     pkg.project_name, specs['~='])
             else:
                 ver, _ = os.path.splitext(specs['~='])
-                result[pkg.key] = '{0}=={1}.*'.format(
-                    pkg.project_name, ver)
+                result[pkg.key] = '{0}>={1},=={2}.*'.format(
+                    pkg.project_name, specs['~='], ver)
```
Comment (greut) [inline requirements_builder/requirements_builder.py]: better indeed! :+1:

File: requirements_builder/requirements_builder.py [hunk 3]
```diff
@@ -39,6 +39,15 @@ def minver_error(pkg_name):
     sys.exit(1)
 
 
+def build_pkg_name(pkg):
+    """Buld package name taking into account extras."""
```
Comment (greut) [inline requirements_builder/requirements_builder.py]: typo and strange wording

File: requirements_builder/requirements_builder.py [hunk 4]
```diff
@@ -39,6 +39,15 @@ def minver_error(pkg_name):
...
+    if pkg.extras:
+        return '{0}[{1}]'.format(
+            pkg.project_name, ','.join(sorted(pkg.extras)))
+    else:
+        return pkg.project_name
```
Comment (greut) [inline requirements_builder/requirements_builder.py]:
```python
if foo:
    return
return
```

**Pull request merged**